src/wp-includes/class-wp-customize-manager.php

```diff
@@ -5012,6 +5019,7 @@ public function customize_pane_settings() {
 			?>
 		</script>
 		<?php
+		wp_print_inline_script_tag( str_replace( array( '<script>', '</script>' ), '', ob_get_clean() ) );
```

Comment:
would it help to create a helper function for this specific task?

```php
/**
 * Removes leading and trailing _empty_ script tags.
 *
 * This is a helper meant to be used for literal script tag construction
 * within wp_print_inline_script_tag(). It removes the literal values of
 * "<script>" and "</script>" from around an inline script.
 *
 * @since 6.4.0
 *
 * @param string $contents Script body with manually created SCRIPT tag literals.
 * @return string Script body without surrounding script tag literals, or
 *                original contents if both exact literals aren't present.
 */
function wp_remove_surrounding_empty_script_tags( $contents ) {
	$opener = '<script>';
	$closer = '</script>';

	$has_both_empty_tags = (
		strlen( $opener ) + strlen( $closer ) > strlen( $contents ) ||
		substr( $contents, 0, strlen( $opener ) ) !== $opener ) ||
		substr( $contents, -strlen( $closer ) ) !== $closer
	);

	return $has_both_empty_tags
		? substr( $contents, strlen( $opener ), -strlen( $closer ) )
		: $contents;
}
```

then these repetitive calls could be marginally nicer. even with the checks for the leading and trailing SCRIPT tags, this should still be faster than `str_replace()`, but more importantly, more secure and resilient against breaking syntax.

```php
$script_body = wp_remove_surrounding_empty_script_tags( ob_get_clean() );
wp_print_inline_script_tag( $script_body );
```

Reply:
There's a syntax error in the `$has_both_empty_tags` condition. Could you fix and add comments explaining the conditions?

Reply:
sure; also just a note - all my examples are almost always wrong and are only meant to convey ideas. if you ever get the impulse, please never copy code of mine from a comment and paste it into a project. don't trust me 😄 I don't test these examples or vet them for WPCS' prefences.

```php
/**
 * Removes leading and trailing _empty_ script tags.
 *
 * This is a helper meant to be used for literal script tag construction
 * within wp_print_inline_script_tag(). It removes the literal values of
 * "<script>" and "</script>" from around an inline script.
 *
 * @since 6.4.0
 *
 * @param string $contents Script body with manually created SCRIPT tag literals.
 * @return string Script body without surrounding script tag literals, or
 *                original contents if both exact literals aren't present.
 */
function wp_remove_surrounding_empty_script_tags( $contents ) {
	$opener = '<script>';
	$closer = '</script>';

	$has_both_empty_tags = (
		strlen( $opener ) + strlen( $closer ) > strlen( $contents ) ||
		substr( $contents, 0, strlen( $opener ) ) !== $opener ||
		substr( $contents, -strlen( $closer ) ) !== $closer
	);

	/*
	 * What should happen if the given contents are not surrounded by
	 * the exact literal script tags? This question opens up a can of
	 * worms with no obvious or clear answer. Removing one of the tags
	 * could lead to just as much trouble as leaving one in place.
	 * This code leaves the string untouched if it can't find both the
	 * exact tags. Maybe someone added an attribute to the opening tag
	 * or maybe someone added a space; it's impossible to know from
	 * here.
	 *
	 * It would be possible to return `null` or `false` here to indicate
	 * the failure, but people probably wouldn't be checking the result
	 * and that could introduce corruption. An empty string could be
	 * another viable return and that would quickly signal that something
	 * is wrong and needs fixing.
	 */
	return $has_both_empty_tags
		? substr( $contents, strlen( $opener ), -strlen( $closer ) )
		: $contents;
}
```

Maybe it's best to return an empty string here if the syntax isn't a perfect match. Is this what you were asking for in "add comments explaining the conditions"?

Reply:
Thanks! I meant more the conditions for `$has_both_empty_tags` as I wasn't understanding exactly they were doing. But it seems that's because the logic wasn't quite right. Also seems worthwhile to normalize the prefix and suffix to upper case. So I believe the condition should actually be:

```php
	$has_both_empty_tags = (
		strlen( $contents ) > strlen( $opener ) + strlen( $closer ) &&
		strtolower( substr( $contents, 0, strlen( $opener ) ) ) === $opener &&
		strtolower( substr( $contents, -strlen( $closer ) ) ) === $closer
	);
```

It has both empty tags if:

1. The entire string is longer than the opener and closer, _and_
2. The string starts with `$opener`, _and_
3. The string ends with `$closer`

When those conditions are satisfied, then `substr( $contents, strlen( $opener ), -strlen( $closer ) )` should be done.

In testing, I see that the first line of the function should also be:

```php
$contents = trim( $contents );
```

In regards to the comment comment before:

```php
	return $has_both_empty_tags
		? substr( $contents, strlen( $opener ), -strlen( $closer ) )
		: $contents;
```

What do you think about actually just doing:

```
if ( ! $has_both_empty_tags ) {
	_doing_it_wrong( __FUNCTION__, '6.4', __( 'Expected string to begin with an empty script tag and close with a script tag.' ) );
	return '';
}
```

I'll note that this is actually the _reverse_ of what `wp_add_inline_script()` does:

https://github.com/WordPress/wordpress-develop/blob/6fa2ce424fddedb4c60fbc90d2aabc9feda8305b/src/wp-includes/functions.wp-scripts.php#L133-L145

Note also how it does the string replacement, so somewhat the naive approach which I originally committed.

Reply:
I've implemented this in https://github.com/WordPress/wordpress-develop/pull/5301

Reply:
> What do you think about actually just doing:

Seems great. There's _never_ a case to call this without the SCRIPT tags, so that makes sense to me. Make it visible as early as possible.

> seems worthwhile to normalize the prefix and suffix to upper case

I have no strong opinion on this. Seems fine. The one thing is that `<sCRiPt>` is not exactly a string literal match, but that's fine. Whatever happens here I think it will work out, and if something goes awry, with the help of the `_doing_it_wrong()` someone will figure it out quickly enough.